### PR TITLE
Provide codesigned stub exe's

### PR DESCRIPTION
**Pull request by Callek**
### Description

- This provides codesigned exe's for conda's stubs
- Relates to [conda/conda #13685](https://github.com/conda/conda/issues/13685)
- Signatures provided on behalf of Anaconda

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
- [x] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: tests/test_codesigned.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        reason = "signtool: unable to locate signtool.exe"
+    try:
+        check_output([signtool, "verify", "/?"])
+    except CalledProcessError as exc:
+        reason = f"signtool: something went wrong while running 'signtool verify /?', output:\n{exc.output}\n"
+    return reason
+
+
+def signtool_unsupported() -> bool:
+    return True if signtool_unsupported_because() else False
```
Comment (schuylermartin45) [inline tests/test_codesigned.py:85]:
```suggestion
    return bool(signtool_unsupported_because())
```

Comment (Callek) [reply tests/test_codesigned.py:85]: I liked the explicitness, but sure ;-)

File: tests/test_codesigned.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+from functools import lru_cache
+from pathlib import Path
+from shutil import which
+from subprocess import CalledProcessError, check_output, run
+
+import pytest
+
+from conda_build.utils import on_win
+
+HERE = os.path.abspath(os.path.dirname(__file__))
```
Comment (schuylermartin45) [inline tests/test_codesigned.py:15]: Should we be mixing `pathlib` and `os.path` calls in the same file? Or do we want to be consistent in usage?

File: tests/test_codesigned.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+REPO_ROOT = (Path(HERE) / "..").resolve().absolute()
+STUB_FOLDER = REPO_ROOT / "conda_build"
```
Comment (schuylermartin45) [inline tests/test_codesigned.py:17]: You could mark these as `Final[Path]` if you wanted to.

**Review by schuylermartin45 — DISMISSED** [review]
Some nit-picky comments but overall I'm glad to see additional testing.

I think the logic for detecting cases where the test can't run looks correct!

File: tests/test_codesigned.py (hunk 2 above)
Comment (Callek) [reply tests/test_codesigned.py:15]: there is no Pathlib variant of shutil.which and other tests don't seem to use Pathlib much at first glance, but I'm a big proponent of Pathlib.

File: tests/test_codesigned.py (hunk 3 above)
Comment (Callek) [reply tests/test_codesigned.py:17]: good point, but unless I need to touch this again before merging I likely won't add Final[Path] to these...

**Comment by Callek** [comment]
> Some nit-picky comments but overall I'm glad to see additional testing.

Spent like 5x as long writing the tests as I did signing the files, but decided we didn't want to be surprised in the future when these are no longer signed.

File: tests/test_codesigned.py (hunk 2 above)
Comment (jezdez) [reply tests/test_codesigned.py:15]: We've been slowly moving paths in conda over to pathlib, but rest assured it backfired in messy code where paths are handed around and assumed strings.

**Review by jezdez — COMMENTED** [review]
I'm not loving that we're adding the same code paths in conda in https://github.com/conda/conda/pull/13721 as well as here. Let's discuss what you're trying to do with these tests. Verify that signtool works?

Also, like https://github.com/conda/conda/pull/13721 this doesn't elaborate how the files were created that we're shipping in every conda package, which is a big miss and should be recorded in this pull request for posterity. Frankly I'm a little concerned that the blobs aren't created in a transparent process via a CD system.

**Comment by jezdez** [comment]
@chenghlee Please take a look at these (which are less urgent than the linked feature in conda) but would still benefit from your seal of approval.

**Review by beeankha — DISMISSED** [review]

**Review by jezdez — DISMISSED** [review]

File: news/5252-sign-stubs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* For Windows users, the stub-exe's that are used for Python entrypoints in packages are now codesigned. (#5252)
```
Comment (jezdez) [inline news/5252-sign-stubs:3]:
```suggestion
* For Windows users, the stub executables used for Python entrypoints in packages are now codesigned. (#5252)
```

**Review by jezdez — APPROVED** [review]

**Review by chenghlee — APPROVED** [review]

**Pull request merged**